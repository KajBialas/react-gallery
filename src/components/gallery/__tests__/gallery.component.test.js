import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import GalleryComponent from '../gallery.component';

describe('GalleryComponent', () => {
  it('should shallows correctly if records are empty', () => {
    const component = shallow(<GalleryComponent />);

    expect(component).toMatchSnapshot();
  });

  it('should shallows correctly if records has data', () => {
    const props = {
      photosRecords: ['1.jpg', '2.jpg'],
    };
    const component = shallow(<GalleryComponent {...props} />);

    expect(component).toMatchSnapshot();
  });
});