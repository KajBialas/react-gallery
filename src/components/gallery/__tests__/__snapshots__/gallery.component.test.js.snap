// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryComponent should shallows correctly if records are empty 1`] = `
<styled.div>
  No result
</styled.div>
`;

exports[`GalleryComponent should shallows correctly if records has data 1`] = `
<styled.div>
  <Styled(MasonryComponent)
    onImagesLoaded={[Function]}
  >
    <styled.div
      key="1.jpg0"
    >
      <Styled(PreloadImage)
        src="1.jpg"
      />
    </styled.div>
    <styled.div
      key="2.jpg1"
    >
      <Styled(PreloadImage)
        src="2.jpg"
      />
    </styled.div>
  </Styled(MasonryComponent)>
</styled.div>
`;
